ts/screens/onboarding/OnboardingNotificationsInfoScreenConsent.tsx

```diff
@@ -0,0 +1,151 @@
+  isLoading,
+  bordered: true
+});
+
+const continueButtonProps = (
+  isLoading: boolean,
+  onPress: () => void
+): BlockButtonProps => ({
+  block: true,
+  onPress,
+  title: I18n.t("onboarding.infoConsent.continue"),
+  isLoading,
+  primary: true,
+  testID: "continue-btn"
+});
+
+const InstructionRow = memo(
+  ({ icon, title }: { icon: React.ReactNode; title: string }) => (
+    <View style={[IOStyles.flex, IOStyles.row, styles.box]}>
+      {icon}
+      <View style={[IOStyles.flex, IOStyles.column, styles.info]}>
+        <H4 weight="Regular" color="black">
+          {title}
+        </H4>
+      </View>
+    </View>
+  )
+);
+
+const OnboardingNotificationsInfoScreenConsent = () => {
+  const dispatch = useIODispatch();
+
+  const optInPreferencesPot = useSelector(profilePreferencesSelector);
+
+  useEffect(() => {
+    const subscription = AppState.addEventListener(
+      "change",
+      async nextAppState => {
+        if (nextAppState === "active") {
+          const authorizationStatus = await checkNotificationPermissions();
+
+          if (authorizationStatus) {
+            dispatch(notificationsInfoScreenConsent());
+          }
+        }
+      }
+    );
+
+    return () => {
+      subscription.remove();
+    };
+  }, [dispatch]);
+
+  const goNext = () => {
+    // When this code executes, we know for sure that system notifications permissions are disabled,
+    // otherwise the component would either have been skipped by the saga or it would have automatically
+    // handled the given permission using the AppState listener (registered on the useEffect)
+
+    if (pot.isSome(optInPreferencesPot)) {
+      const optInPreferences = optInPreferencesPot.value;
+      if (optInPreferences.reminder) {
+        void mixpanelTrack("NOTIFICATIONS_OPTIN_REMINDER_ON_PERMISSIONS_OFF");
+      }
+    }
+
+    dispatch(notificationsInfoScreenConsent());
+  };
+
+  const openSettings = () => {
+    openAppSettings();
+  };
+
+  return (
+    <BaseScreenComponent
+      headerTitle={I18n.t("onboarding.infoConsent.headerTitle")}
+      contextualHelp={emptyContextualHelp}
+      goBack={false}
+      customGoBack={<View />}
+    >
+      <SafeAreaView style={IOStyles.flex}>
+        <ScreenContent
+          title={I18n.t("onboarding.infoConsent.title")}
+          subtitle={I18n.t("onboarding.infoConsent.subTitle")}
+        >
+          <View style={styles.container}>
+            <InstructionRow
+              icon={<NotificationsIcon width={44} height={44} />}
+              title={I18n.t("onboarding.infoConsent.instruction1")}
+            />
+            <View style={styles.separator} />
+            <InstructionRow
+              icon={<NotificationsToggleIcon width={44} height={23} />}
+              title={I18n.t("onboarding.infoConsent.instruction2")}
+            />
+          </View>
+        </ScreenContent>
+        <FooterWithButtons
+          type="TwoButtonsVertical"
```

Comment:
I leave this comment here just as a starting point:
we actually we don't need to customize the `FooterWithButtons` component since we already have a component to cover this scenario: [`FooterStackButtons`](https://github.com/pagopa/io-app/blob/98be863bb7e980e7ee26bb4a6cf7946e707c11ef/ts/features/bonus/bonusVacanze/components/buttons/FooterStackButtons.tsx#L57)

We can remove the changes on `FooterWithButtons` component, what do you think?

Reply:
changed https://github.com/pagopa/io-app/pull/4147/commits/bdbb4872c39f4435e8850880b1fa5551228f6627
It might be useful to move this component to `/ts/components/ui`, what do you think?